### ISSUE TITLE
Add automatic HTML encoding of unallowed unicode characters

### ISSUE THREAD

**Issue by Merudo** [RPTools/maptool#1603]
**Is your feature request related to a problem? Please describe.**
A number of unicode characters are frequently used by other languages.

Examples: é, è, ë, ç, ¿, µ, etc.

Current MapTool behavior is to produce an error message on any use of these symbols in MapTool functions.

Instead of generating an error, it would be more convenient for users if these symbols were automatically encode.

So, for example entering 

`[h: notes = json.set("{}", "Classe", "Prêtre")]`

 would automatically replace `ê` by `&#234`, which would give the same result as

`[h: notes = json.set("{}", "Classe", "Pr&#234tre")]`

**Describe the solution you'd like**
No automatically encoding those symbols.

**Additional context**
The now defunct Mote managed to do it. Maybe we can draw inspiration from that.

**Comment by Phergus** [comment]
I'm sure a lot of folks would appreciate this.

Could it be done just before the macro code is passed to the parser?

Impact on doing lookups/searches in lists or arrays from some user input passed from a form() or input()?

**Comment by Azhrei** [comment]
It would also mean people that copy/paste code from a word processor (instead of an editor) would immediately find out that they used the wrong single or double quotes.  That's +1 as well.

**Comment by Merudo** [comment]
I just realized, we might not even have to encode the characters?

I checked and once a string containing problematic characters is inside a MapTool property, the string can be used pretty much everywhere (in `json`, in other macros, etc).

Seems most of the software is fine with accents and whatnot. Only barrier is the first reading of the content.

**Comment by Azhrei** [comment]
Yes, ANTLR v2 isn’t 8-bit clean.  At least, that’s what I remember about the issue...

**Comment by Phergus** [comment]
#1943 adds ISO-8859-1 char support
